core/src/main/java/io/grpc/internal/GrpcUtil.java

```diff
@@ -582,15 +586,37 @@ public Long parseAsciiString(String serialized) {
    */
   @Nullable
   static ClientTransport getTransportFromPickResult(PickResult result, boolean isWaitForReady) {
-    ClientTransport transport;
+    final ClientTransport transport;
     Subchannel subchannel = result.getSubchannel();
     if (subchannel != null) {
       transport = ((SubchannelImpl) subchannel).obtainActiveTransport();
     } else {
       transport = null;
     }
     if (transport != null) {
-      return transport;
+      final ClientStreamTracer.Factory streamTracerFactory = result.getStreamTracerFactory();
+      if (streamTracerFactory != null) {
```

Comment:
Actually small nit: maybe for better readability, make this an early return? (`if (streamTracerFactory == null) { return transport; }`)